dp-cluster-setup-utility.py

```diff
@@ -882,6 +893,143 @@ def streamsmsgmgr_url(self):
   def host_name(self, service_name, component_name):
     return self.ambari.cluster.service(service_name).component(component_name).host_names()[0]
 
+
+class BeaconProxyTopology:
+  TEMPLATE = """<?xml version="1.0" encoding="UTF-8"?>
+  <topology>
+     <uri>{knox_url}/gateway/beacon-proxy</uri>
+     <name>beacon-proxy</name>
+     <timestamp>{timestamp}</timestamp>
+     <generated>false</generated>
+     <gateway>
+        <provider>
+          <role>authentication</role>
+          <name>HadoopAuth</name>
+          <enabled>true</enabled>
+          <param>
+            <name>config.prefix</name>
+            <value>hadoop.auth.config</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.signature.secret</name>
+            <value>knox-signature-secret</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.type</name>
+            <value>kerberos</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.simple.anonymous.allowed</name>
+            <value>false</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.token.validity</name>
+            <value>1800</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.cookie.domain</name>
+            <value>HWQE.HORTONWORKS.COM</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.cookie.path</name>
+            <value>gateway/beacon-proxy</value>
+          </param>
+          <param>
+            <name>hadoop.auth.config.kerberos.principal</name>
+            <value>HTTP/{knox_host}@HWQE.HORTONWORKS.COM</value>
```

Comment:
Is this hardcoded realm name always the same?

Reply:
Thanks for pointing this out. But i am not sure how to figure out the real from ambari. Would like to hear your suggestion.

Reply:
You should be able to get it from kerberos-env/realm.

Try something like

```python
cluster.config_property('kerberos-env', 'realm')
```